cmd/helm/dependency_update.go

```diff
@@ -70,13 +72,31 @@ func newDependencyUpdateCmd(out io.Writer) *cobra.Command {
 			if client.Verify {
 				man.Verify = downloader.VerifyAlways
 			}
-			return man.Update()
+
+			err := man.Update()
+			if err != nil {
+				return err
+			}
+
+			if client.Untar {
+				match := chartpath + "/charts/*.tgz"
+				files, err := filepath.Glob(match)
+				if err != nil {
+					return err
+				}
+				for _, f := range files {
+					chartutil.ExpandFile(chartpath+"/charts/", f)
+					os.Remove(f)
+				}
+			}
+			return nil
```

Comment:
I understand why you have had to implement the untar capability here in the CLI package due to it calling the download manager update directly. The problem with this is that because it is not implemented in the library package ('pkg') then it is only available when using the Helm CLI. 

This would mean implementing  it in the [download manager](https://github.com/helm/helm/blob/master/pkg/downloader/manager.go). From looking at it, I think that this would unfortunately break backwards compatibility. This seems like a "catch 22".

Reply:
As you said, I move it to  `pkg/downloader`, and I think it is more appropriate. 
At the same time, it also brings me a puzzle. @mattfarina  [suggests](https://github.com/helm/helm/issues/8479#issuecomment-661073885) that I refer to the implementation of `helm pull`, which is also treated as a cli behavior. What is the direction of our code evolution? Is there any relevant roadmap for reference?

To this end, I keep two implementations: in [cmd/helm](https://github.com/helm/helm/pull/8499/commits/ea84849511b9cbd5bcbe8578f84f901c0ffea703) and [pkg/downloader](https://github.com/helm/helm/pull/8499/commits/1c46fddca590fb905cce26aa2fde941ef4c05c03). To explore the best implementation.